web3/eth.py

```diff
@@ -374,6 +374,7 @@ def contract(self,
     def setContractFactory(self, contractFactory):
         self.defaultContractFactory = contractFactory
 
+    @deprecated_for("Dropped in EIP 1474")
```

Comment:
I think the message a user will get is going to be awkward

https://github.com/ethereum/web3.py/blob/e578dd266ecff2e01398407cd6472894ddb59bc6/web3/utils/decorators.py#L54

This probably warrants just manually raising the proper deprecation warning exception.  Similarly, in v5, we should potentially not fully remove the methods, but make them do a `raise DeprecationWarning()` which *I believe* that when raised as an exception is just like normal exception raising.  This way people aren't greeted with an `AttributeError` but a nice explanation of why the method they are accessing is missing.

Reply:
So warnings don't break codeflow, they just dump information out so a user who's watching for them knows something is going away.  Using the `raise` syntax breaks code flow so they are two separate mechanisms.  These can be easy to mixup since the exception class is called `DeprecationWarning` but they are to different mechanisms that use the same class.

- `warning.warn(...)` - for `v4`
- `raise DeprecationWarning`  - for `v5`

Reply:
Are we switching to the `raise DeprecationWarning` for all methods we're deprecating in v5? Or just specific ones?

Reply:
`raise DeprecationWarning` should only be used if it was *previously* deprecated meaning that users of `v4` would have already been warned.

Otherwise it should use `warning.warn(...)` to begin giving users a warning